### DISCUSSION TITLE
The result of the code execution is not saved?

### DISCUSSION THREAD

**Discussion by bestpika** [alagrede/znote-app#67] [Q&A]
I found that the result of the executed code is not displayed directly when I open the file next time, I have to run it again?
If so, is it possible to add a function for saving results?
After all, this is what writing notes is all about.

**Comment by alagrede** [comment]
Hi @bestpika,
I'm agree with you, but showing results on open is a tough security issue for me 😅 
The goal is to prevent users from opening an untrusted note and showing malicious JS results.
I guess I could add a trust mechanism popup to allow showing results. Anyway, to do this, I have to be careful to keep a good level of security in this application for all users.

**Reply by bestpika** [reply]
I agree with your concerns about security.
But I am a bit confused as to which situation you are referring to.
The "result" itself should not be a security issue, but from your description, it seems to mean that it is automatically executed once when the file is opened?

**Reply by alagrede** [reply]
No. The results are not executed when the file is opened. But saving the results and displaying them on opening would mean that the results will be executed on opening and that's my security concern. 
Results can contains HTML but also JS (dynamic graph, sortable table, React app with listener...) so, it is not sufficient to save the HTML results. That's why I'm talking about a trusted popup mechanism (like VScode actually) to execute only known code.
